torchao/float8/float8_scaling_utils.py

```diff
@@ -36,6 +36,7 @@ def hp_tensor_to_float8_dynamic(
     linear_mm_config: LinearMMConfig,
     reduce_amax: bool = False,
     gemm_input_role: GemmInputRole = GemmInputRole.INPUT,
+    device_mesh = None,
```

Comment:
DeviceMesh only exists in newer pytorch (2.4+). I tried to annotate device_mesh with string `"torch.distributed.DeviceMesh"`. but some CI run torch older pytorch 2.2 and complain about not being able to find DeviceMesh. I gave up on type hint. hopefully device_mesh itself is a strong indicator of DeviceMesh

Reply:
for now, we are intending to only test float8 code on the latest nightly to move fast and minimize maintenance load.  Do you know CI is giving you issues?  We have disabled CI on older PT versions per-test-file using version checks.

Reply:
I belive it's the following one. My guess - the function signature is checked at import time. For torch==2.2.2, if one of the test imported `float8_utils.py` or `float8_scaling_utils.py`, it will throw error

```
[test (CUDA 2.2.2, linux.g5.12xlarge.nvidia.gpu, torch==2.2.2 "numpy<2" , cuda, 12.1) / linux-job](https://github.com/pytorch/ao/actions/runs/11023026757/job/30613550360?pr=933#logs)
```